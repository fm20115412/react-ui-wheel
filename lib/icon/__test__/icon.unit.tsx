import rerender from 'react-test-renderer'
import Icon from '../icon'
import React from 'react'
import { mount } from 'enzyme'

describe('测试icon', () => {
    it('这是个icon', () => {
        let json = rerender.create(<Icon name='wechat' />).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('它可以点击', () => {

        let fn = jest.fn()
        let componnet = mount(<Icon name='baidu' onClick={fn} />)
        componnet.find('svg').simulate('click')
        expect(fn).toBeCalled()
    })
})