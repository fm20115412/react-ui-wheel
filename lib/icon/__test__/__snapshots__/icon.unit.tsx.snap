// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试icon 这是个icon 1`] = `
<svg
  className="icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
